Lib/Database/MySqlQuery.php

```diff
@@ -0,0 +1,1247 @@
+<?php
+/**
+ * MySqlQuery class file.
+ *
+ * @package Freyja\Database
+ * @copyright 2016 SqueezyWeb
+ * @author Gianluca Merlo <gianluca@squeezyweb.com>
+ * @since 0.1.0
+ */
+
+namespace Freyja\Database;
+
+use Freyja\Exceptions\InvalidArgumentException as InvArgExcp;
+use RuntimeException;
+
+/**
+ * MySqlQuery class.
+ *
+ * @package Freyja\Database
+ * @author Gianluca Merlo <gianluca@squeezyweb.com>
+ * @since 0.1.0
+ * @version 1.0.0
+ */
+class MySqlQuery extends Query implements QueryInterface {
+  /**
+   * Query type.
+   *
+   * Accepted values: 'select', 'update', 'insert', 'delete'.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var string
+   */
+  private $type;
+
+  /**
+   * Target table.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var string
+   */
+  private $table;
+
+  /**
+   * SELECT fields.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var array
+   */
+  private $select = array();
+
+  /**
+   * COUNT.
+   *
+   * True correspond to `COUNT(*)`. Or array containing the fields to be
+   * counted. Default false.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var boolean|array
+   */
+  private $count = false;
```

Comment:
Is there a reason why this defaults to `false`?
When it is not initialized, but only declared, you can still check it like so:
```php
if ($this->count)
  // Do stuff...
```